tsl/src/partialize_finalize.c

```diff
@@ -169,8 +250,35 @@ inner_agg_deserialize(FACombineFnMeta *combine_meta, bytea *serialized_partial,
 
 		FC_ARG(deser_fcinfo, 0) = PointerGetDatum(serialized_partial);
 		FC_NULL(deser_fcinfo, 0) = serialized_isnull;
-		combine_meta->deserialfn_fcinfo->isnull = false;
-		deserialized = FunctionCallInvoke(deser_fcinfo);
+		deser_fcinfo->isnull = false;
+
+		/*
+		 * When an exception is thrown and longjmp() is called, CurrentMemoryContext is potentially
+		 * different than what it was inside the PG_TRY() block below.
+		 *
+		 * Restore it to the old value so that the code in the subsequent PG_CATCH() block does not
+		 * corrupt the memory.
+		 *
+		 * No need for volatile variables since we don't modify any of this function's stack frame
+		 * inside the PG_TRY() block.
+		 */
+		MemoryContext oldcontext = CurrentMemoryContext;
+		PG_TRY();
+		{
+			deserialized = FunctionCallInvoke(deser_fcinfo);
```

Comment:
Do we know the length of the incoming value, and what it should be for the old vs new format?
If so, I'd decide based on the length here. Doing a failed function call with all its setjmp/longjmp for each row is probably slow...

Reply:
Ah, numeric serialization is variable length, that's why.

Reply:
Deciding based on extra length looks safe, because at least we know the total serialized length when we read it, and the difference is that two 8-byte counts of +- infinities were added to the end. Numeric serialization itself haven't changed, so shouldn't be possible to accidentally read the old data in the new format.

Reply:
The problem is that some of the stored partials are of the old format, and some of the new partials are of the new format. We can't identify what extra length is without diving into the internals of the numeric deserializer function. It is theoretically possible though.

Reply:
I profiled the performance for vanilla timescaledb/PG14 vs this PR/PG14 with normal length partials and the performance was identical despite the `PG_TRY()` block being there, which is the one that calls `setjmp`.